lib/ansible/plugins/action/nxos_file_copy.py

```diff
@@ -350,27 +345,41 @@ def process_outcomes(session, timeout=None):
         # Spawn pexpect connection to NX-OS device.
         nxos_session = pexpect.spawn('ssh ' + nxos_username + '@' + nxos_hostname + ' -p' + str(port))
         # There might be multiple user_response_required prompts or intermittent timeouts
-        # spawning the expect session so loop up to 5 times during the spwan process.
-        for connect_attempt in range(6):
+        # spawning the expect session so loop up to 24 times during the spawn process.
+        max_attempts = 24
```

Comment:
Note, on another issue, other members of the core team recently talked about wanting retries to be user settable (so in this case, probably a parameter to the action plugin).  I'm not sure if they would consider this something to do in this case or not but you probably should ask for guidance.

Since this was already retrying without user control over the number of times, I'm going to merge this backport, though.

Reply:
@abadger Not a bad idea for sure.  We can make this change for `2.10` to make it user configurable with a reasonable default setting.